libs/framework/src/module.c

```diff
@@ -454,10 +452,7 @@ static celix_status_t celix_module_loadLibrariesInManifestEntry(celix_module_t*
 
         if ( (status == CELIX_SUCCESS) && (activator != NULL) && (strcmp(trimmedLib, activator) == 0) ) {
             *activatorHandle = handle;
-        } else if ((status != CELIX_SUCCESS) && (handle != NULL)) {
-            celix_libloader_close(fwCtx, handle);
```

Comment:
Note this is dead code, i.e. when handle != NULL we must have status == CELIX_SUCCESS.